pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -79,10 +78,32 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.
 		return res, err
 	}
 
-	// setup dispatcher for vclusterops API
-	dispatcher, err := q.makeDispatcher(q.Log, q.Vdb, nil /*password*/)
+	// check version for Vdb, the minimim version should be 24.2.0
+	vinf, err := q.Vdb.MakeVersionInfoCheck()
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, nil
+	}
+	if !vinf.IsEqualOrNewer(v1.RestoreSupportedMinVersion) {
+		q.VRec.Event(q.Vrpq, corev1.EventTypeWarning, events.IncompatibleDB, "Incompatibility with the database")
+		err = vrpqstatus.Update(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+			[]*metav1.Condition{v1.MakeCondition(vapi.QueryComplete, metav1.ConditionTrue, "IncompatibleDB")}, stateFailedQuery, nil)
```

Comment:
I don't think it makes sense to set `QueryComplete` when `Querying has not started. Also I do not think we want to retry if version does not support Restore or vclusterops is disabled. So maybe we can add an extra condition that indicates that meet all the requirements to successfully perform a `Query` and only when it is true can we start`Querying`. Or we can add a special reason to `Querying` to express what I just described. What do you think @spilchen ?

Reply:
I agree. We shouldn't retry if things aren't compatible because chances are they will be incompatible next reconcile iteration too. 

Maybe add a QueryReady status condition. We can then set this to false, with a reason depending on what we find.